### PR TITLE
49586 adjust the diagnostics dialog towards filing a bug vs downloading diagnostics

### LINKED ISSUE #49586: Adjust the diagnostics dialog towards filing a bug (vs downloading diagnostics)
**Context**
The current diagnostics dialog is oriented towards downloading (and reviewing) a JSON file containing diagnostics, which are eventually useful to file a bug.

We'd like this dialog to be about filing a bug that contains the diagnostics information, along with extra context users could add themselves, so the bug could be more easily filled.

**Current dialog**
Image

**Proposal**
Update the dialog's title to "Report a bug"
Add a "Context" free form text area with a placeholder "Please describe your issue and steps to help us reproduce it"
Place the diagnostics option below, possibly in a toggled area to ensure most context remain ticked by default
For now keep the "Download" action the main one for end users

### PR DESCRIPTION
**Pull request by filiphric**
Closes https://github.com/metabase/metabase/issues/49586

### Description

Design updates to diagnostic window. When slack bug reporting channel is set, modal is shows a "description" field. Modal now shows a "bug report submitted" state too.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. enable `MB_BUG_REPORTING_ENABLED` env setting
2. hit F1 or choose any other way of showing bug report modal

### Demo

Screenshot 2024-11-20 at 15 10 30
Screenshot 2024-11-20 at 15 10 27
Screenshot 2024-11-20 at 15 09 18
Screenshot 2024-11-20 at 15 05 43

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/components/ErrorPages/ErrorDiagnosticModal.tsx [hunk 1]
```diff
@@ -90,58 +189,123 @@ export const ErrorDiagnosticModal = ({
...
+                  </Text>
+                  <Text color="text-medium" w="80%">
+                    {t`This information helps ${applicationName} figure out what exactly caused the issue.`}
+                  </Text>
+                </Stack>
+                {enableBugReportField && (
+                  <Button
+                    variant="subtle"
+                    p="0"
+                    style={{ fontSize: "17px", width: "50px" }}
```
Comment (iethree) [inline frontend/src/metabase/components/ErrorPages/ErrorDiagnosticModal.tsx:221]: there are [mantine style props](https://v6.mantine.dev/styles/style-props/) for these! `fz` and `w`

I don't think, though that you necessarily want to set a fixed width, it causes weird ellipsification on some screen sizes, and will probably truncate other languages strangely.


Screen Shot 2024-11-21 at 8 11 59 AM

**Review by iethree — CHANGES_REQUESTED** [review]
Looking good! I like how this is shaping up. Some comments inline, but higher level:

1. This was a little hard to review, because this PR seems to be based on top off https://github.com/metabase/metabase/pull/49875 and https://github.com/metabase/metabase/pull/49954, so it's hard to tell which parts of the diff are actually part of this PR, you can change the `base` branch in github to fix this up
3. The backend seems to be dropping fields on their way to slack 😞

**Comment by mazameli** [comment]
@filiphric, could you help @romnitt to give this a close UX review (if he needs help getting this to run locally, or seeing an ephemeral PR env with this branch etc)?

**Review by iethree — CHANGES_REQUESTED** [review]
Thanks for restructuring those modals, they're much easier to understand now. I still think there's a problem with sending the `description` field to slack. This is the test I just ran


Screen Shot 2024-11-26 at 5 47 21 PM
Screen Shot 2024-11-26 at 5 47 13 PM

I also don't seen any messages in the bot-testing or bug-reports-internal slack channels that actually have a description. they're all `N/A`

**Review by uladzimirdev — APPROVED** [review]
FE part LGTM, can we get an approval from @romnitt as @mazameli asked?

**Review by johnswanson — APPROVED** [review]

**Review by escherize — APPROVED** [review]

**Review by iethree — APPROVED** [review]

**Pull request closed without merging**